src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -233,7 +233,14 @@ public string GetNodeLocation()
 
             string nodeFolder = NodeHandler.nodeFolder;
 
-            if (taskHasNode20Data)
+            if (taskHasNode20Data && !isNode20SupportedSystems())
```

Comment:
>isNode20SupportedSystems [](http://example.com/codeflow?start=38&length=24)

would be good to add a knob for this to 'force' allowing tasks to run on node20 runner of "isNode20SupportedSystems"

Reply:
We already have one to force agent to use node16: AGENT_USE_NODE16, this one might already serve for this purpose?
https://github.com/microsoft/azure-pipelines-agent/blob/a3067b75926acec16ef0166ca34c3f6ad4e1ba82/src/Agent.Sdk/Knob/AgentKnobs.cs#L148C1-L153C52